covalent_ui/webapp/src/components/settings/SettingsCard.js

```diff
@@ -0,0 +1,1062 @@
+  const [tempDataServer, setTempDataServer] = useState(null)
+
+  useEffect(() => {
+    dispatch(settingsResults())
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (valueChange) {
+      const settingObj = {
+        isChanged: valueChange,
+        data: resultOutput,
+        nodeName: resultKey,
+        mainKey: accName,
+      }
+      dispatch(toggleLatticeDrawer(settingObj))
+    } else {
+      const settingObj = {
+        isChanged: valueChange,
+      }
+      dispatch(toggleLatticeDrawer(settingObj))
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [valueChange, resultOutput])
+
+  useEffect(() => {
+    if (settings_result) {
+      setSearchData(Object.values(settings_result)[0])
+      setClientDetail(Object.values(settings_result)[0])
+      setServerDetail(Object.values(settings_result)[1])
+      setTempData(Object.values(settings_result)[0])
+      setTempDataServer(Object.values(settings_result)[1])
+    }
+  }, [settings_result])
+
+  useEffect(() => {
+    if (_.size(searchData) !== 0) {
+      setResultOutput(Object.values(searchData)[0])
+      setRestoreData(Object.values(searchData)[0])
+    }
+  }, [searchData])
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    const updateData = {
+      [accName]: {
+        [resultKey]: resultOutput,
+      },
+    }
+    dispatch(updateSettings(updateData)).then((action) => {
+      if (action.type === updateSettings.fulfilled.type) {
+        setOpenSnackbar(true)
+        setSnackbarMessage('Settings Updated Successfully')
+        setValueChange(false)
+        setClientDetail((currValue) => ({
+          ...currValue,
+          [resultKey]: resultOutput,
+        }))
+        setRestoreData(resultOutput)
+      } else if (action.type === updateSettings.rejected.type) {
+        setOpenSnackbar(true)
+        setSnackbarMessage(
+          'Something went wrong and could not settings updated!'
+        )
+      }
+    })
+    setHandle('')
+  }
+
+  const getSubmenuName = (name) => {
+    let formattedName = name
+    const uSpliit = name.includes('_')
+    if (uSpliit) {
+      var a = name
+        .replace(name.at(0), name.at(0).toLocaleUpperCase())
+        .substring(0, name.indexOf('_'))
+      var b = name
+        .replace(name.at(0), name.at(0).toLocaleUpperCase())
+        .substring(name.indexOf('_') + 1, name.length)
+      formattedName = a + ' ' + b.replace(b.at(0), b.at(0).toLocaleUpperCase())
+    } else {
+      if (name === 'slurm') {
+        formattedName = name.toUpperCase()
+      } else if (name === 'dask') {
+        formattedName = name.toUpperCase()
+      } else {
+        formattedName = name.charAt(0).toUpperCase() + name.slice(1)
+      }
+    }
+    return formattedName
+  }
+
+  const getLabelName = (name) => {
+    let formattedName = name
+    const uSpliit = name.includes('_')
+    if (uSpliit) {
```

Comment:
@Prasy12 there's an issue with this logic treating for example: `threads_per_worker` would end up having a formattedName of `Threads per_worker`, also there is code duplication of this doing the same thing in `getSubmenuName`.

I would suggest the following code is used to handle that case:

```
const formatUnderscoreConcatenatedString = (str) => {
    return str.split('_').map(word => word.charAt(0).toUpperCase() + word.slice(1)).join(" ")
  }
```

And this be re-used in `getLabelName` and `getSubmenuName`.

Also I would ask that variable naming be improved as `a` and `b` aren't very verbose.